torch/quantization/fx/_equalize.py

```diff
@@ -0,0 +1,221 @@
+import torch
+import torch.nn as nn
+from torch.quantization.observer import PerChannelMinMaxObserver
+
+import warnings
+
+
+class _InputEqualObserver(nn.Module):
+    r"""Observer for tracking the running min/max values of input columns, and
+    computing the quantization parameters for the overall min/max input values.
+
+    Args:
+        dtype: Quantized data type
+        qscheme: Quantization scheme
+        quant_min: Minimum quantization value. If unspecified, it will
+            follow the 8-bit setup.
+        quant_max: Maximum quantization value. If unspecified, it will
+            follow the 8-bit setup.
+
+    The running minimum/maximum :math:`x_\text{min/max}` are computed in the
+    same way as :class:`~torch.quantization.observer.PerChannelMinMaxObserver`,
+    with the difference that the running min/max values are stored per column.
+
+    The qparams are calculated by multiplying the min/max input column values
+    with the equalization scale, reducing to find the global min/max input
+    values, and then calculating in the same way as in
+    :class:`~torch.quantization.observer.MinMaxObserver`
+
+    .. note:: If the running minimum equals to the running maximum, the scales
+              and zero_points are set to 1.0 and 0.
+    """
+
+    def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
+                 quant_min=None, quant_max=None, factory_kwargs=None) -> None:
+        super(_InputEqualObserver, self).__init__()
+
+        if qscheme not in {torch.per_tensor_affine, torch.per_tensor_symmetric}:
+            raise TypeError("Input qscheme must be per-tensor")
+
+        self.input_obs = PerChannelMinMaxObserver(ch_axis=1, dtype=dtype,
```

Comment:
I think we should have an output_obs as well for output observer, right?

Reply:
I thought the output observer was taken into account in the from_float() function and we would just replace the InputEqualizationObserver with a MinMaxObserver in prepare()?

Reply:
The output observer is set to whatever is used for the input observer currently. So if the user wants to use something other than MinMax for the output, they wouldn't be able to.
We can add an optional argument to the InputEqualizationObserver that takes in output_observer and set it to default MinMaxObserver.

Reply:
Oh ok, got it!